packages/db/src/lib/granule.ts

```diff
@@ -40,18 +40,27 @@ export const getGranuleCollectionId = async (
  *   Granules/executions PG model class instance
  * @returns {Promise<PostgresGranuleRecord[]>}
  */
-export const upsertGranuleWithExecutionJoinRecord = async (
-  knexTransaction: Knex.Transaction,
-  granule: PostgresGranule,
-  executionCumulusId?: number,
+export const upsertGranuleWithExecutionJoinRecord = async ({
+  knexTransaction,
+  granule,
+  executionCumulusId,
   granulePgModel = new GranulePgModel(),
-  granulesExecutionsPgModel = new GranulesExecutionsPgModel()
-): Promise<PostgresGranuleRecord[]> => {
-  const [pgGranule] = await granulePgModel.upsert(
-    knexTransaction,
-    granule,
-    executionCumulusId
-  );
+  granulesExecutionsPgModel = new GranulesExecutionsPgModel(),
+  writeConstraints = true,
+}: {
+  knexTransaction: Knex.Transaction;
+  granule: PostgresGranule;
+  executionCumulusId?: number;
+  granulePgModel?: GranulePgModel;
+  granulesExecutionsPgModel?: GranulesExecutionsPgModel;
+  writeConstraints?: boolean;
```

Comment:
This function had to be parameterized to allow for the branch conditional behavior.     Many of the updates around the use of this function are solely to accommodate that change.